model/itemTab.cc

```diff
@@ -282,23 +287,27 @@ namespace minsky
                         { // draw vertical grid lines
                           cairo::CairoSave cs(cairo);
                           cairo_set_source_rgba(cairo,0,0,0,0.5);
+                          y1=it==itemVector[0]? 0.5*rowHeight: 0;
                           for (x=x0; x<w+colWidth; x+=colWidth)
                             {
                               cairo_move_to(cairo,x,y-2*rowHeight);
-                              cairo_line_to(cairo,x,y+0.5*rowHeight);
+                              cairo_line_to(cairo,x,y+y1);
                               cairo_stroke(cairo);
                             }
-                        }                                            
-                        { // draw horizontal grid line
-                          cairo::CairoSave cs(cairo);
-                          cairo_set_source_rgba(cairo,0,0,0,0.5);
-                          cairo_move_to(cairo,x0,y0-0.5*rowHeight);
-                          cairo_line_to(cairo,w+colWidth,y0-0.5*rowHeight);
-                          cairo_stroke(cairo);
-                        }                                  
+                        }
+                        
+                        if (it==itemVector[0])                                            
+                          { // draw horizontal grid line
+                            cairo::CairoSave cs(cairo);
+                            cairo_set_source_rgba(cairo,0,0,0,0.5);
+                            cairo_move_to(cairo,x0,y0-0.5*rowHeight);
+                            cairo_line_to(cairo,w+colWidth,y0-0.5*rowHeight);
+                            cairo_stroke(cairo);
+                          }                                  
                         cairo::CairoSave cs(cairo);
                         // make sure rectangle has right height
-                        cairo_rectangle(cairo,x0,y0-1.5*rowHeight,w+colWidth,y-y0+2*rowHeight);    
+                        if (it==itemVector[0]) cairo_rectangle(cairo,x0,y0-1.5*rowHeight,w+colWidth,y-y0+2*rowHeight);    
+                        else cairo_rectangle(cairo,x0,y0-rowHeight,w+colWidth,y-y0+rowHeight);    
```

Comment:
Could we not reuse the previous y1 calculation here?

Reply:
Done :)